### DISCUSSION TITLE
Getting Error on this step "python -m local.ingest"

### DISCUSSION THREAD

**Discussion by indudfx** [truefoundry/cognita#198] [Q&A] [answered]
This is the error message I get when i try to run "python -m local.ingest" module

requests.exceptions.MissingSchema: Invalid URL '/embeddings': No scheme supplied. Perhaps you meant https:///embeddings?

what could be problem, please help me

**Comment by HariHaranDFX** [comment]
I am getting same, please suggest what can be done

DEBUG:    2024-06-02 17:48:21,524 - qdrant:get_collections:166 - [Qdrant] Fetching collections
DEBUG:    2024-06-02 17:48:21,524 - qdrant:get_collections:168 - [Qdrant] Fetched 0 collections
DEBUG:    2024-06-02 17:48:21,524 - qdrant:create_collection:43 - [Qdrant] Creating new collection docs
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Hari\MyProjects\Python\cognita\local\run.py", line 3, in <module>
    from backend.modules.query_controllers import BasicRAGQueryController
  File "C:\Hari\MyProjects\Python\cognita\backend\modules\query_controllers\__init__.py", line 1, in <module>
    from backend.modules.query_controllers.example.controller import BasicRAGQueryController
  File "C:\Hari\MyProjects\Python\cognita\backend\modules\query_controllers\example\controller.py", line 18, in <module>
    from backend.modules.metadata_store.client import METADATA_STORE_CLIENT
  File "C:\Hari\MyProjects\Python\cognita\backend\modules\metadata_store\client.py", line 4, in <module>
    METADATA_STORE_CLIENT = get_metadata_store_client(config=settings.METADATA_STORE_CONFIG)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Hari\MyProjects\Python\cognita\backend\modules\metadata_store\base.py", line 212, in get_metadata_store_client
    return METADATA_STORE_REGISTRY[config.provider](config=config.config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Hari\MyProjects\Python\cognita\backend\modules\metadata_store\local.py", line 69, in __init__
    VECTOR_STORE_CLIENT.create_collection(
  File "C:\Hari\MyProjects\Python\cognita\backend\modules\vector_db\qdrant.py", line 46, in create_collection
    partial_embeddings = embeddings.embed_documents(["Initial document"])
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Hari\MyProjects\Python\cognita\backend\modules\embedder\embedding_svc.py", line 44, in embed_documents
    return self.call_embedding_service(texts, "documents")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Hari\MyProjects\Python\cognita\backend\modules\embedder\embedding_svc.py", line 38, in call_embedding_service
    response = requests.post(self.url.rstrip("/") + "/embeddings", json=payload)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Hari\MyProjects\Python\cognita\.venv\Lib\site-packages\requests\api.py", line 115, in post
    return request("post", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Hari\MyProjects\Python\cognita\.venv\Lib\site-packages\requests\api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Hari\MyProjects\Python\cognita\.venv\Lib\site-packages\requests\sessions.py", line 575, in request
    prep = self.prepare_request(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Hari\MyProjects\Python\cognita\.venv\Lib\site-packages\requests\sessions.py", line 484, in prepare_request
    p.prepare(
  File "C:\Hari\MyProjects\Python\cognita\.venv\Lib\site-packages\requests\models.py", line 367, in prepare
    self.prepare_url(url, params)
  File "C:\Hari\MyProjects\Python\cognita\.venv\Lib\site-packages\requests\models.py", line 438, in prepare_url
    raise MissingSchema(
requests.exceptions.MissingSchema: Invalid URL '/embeddings': No scheme supplied. Perhaps you meant https:///embeddings?

**Comment by agutgutia1991** [comment]
Hi All,
We will get back on these soon. Thanks for reaching out.
Thanks,
Anuraag

**Comment by S1LV3RJ1NX** [comment, marked as answer]
Apologies for inconvenience, we were in process of updating the docs and making local setup quick and easy. The README is now updated. You need an Embedding service such as [Infinity](https://github.com/michaelfeil/infinity) for ingestion to work. We have added the instructions in the README. 

You can now use the [docker-compose](https://github.com/truefoundry/cognita?tab=readme-ov-file#whale-using-docker-compose-recommended) where everything is preset for you, this will automatically run the ingestion and open up the frontend.

If you want to use, locally (a bit of hustle, but if you want to :P), you need to setup the [infinity-service](https://github.com/truefoundry/cognita?tab=readme-ov-file#infinity-service).
Create a copy of `compose.env` file as `.env` [file](https://github.com/truefoundry/cognita#setting-up-env-file) and populate the environment vars `EMBEDDING_SVC_URL` & `RERANKER_SVC_URL` with the url of the service.  Then run the ingestion command.

**Reply by HariHaranDFX** [reply]
Thanks for the help @S1LV3RJ1NX 

I've successfully built my project using docker-compose, and all components seem to be running smoothly. However, I've encountered a problem with the frontend: the lists in the add datasource and collection datas are not being populated, despite the backend API and Infinity server API functioning correctly.

I've attempted to manually connect the frontend to the backend API, but it doesn't seem to be reflecting the changes. I'm unsure where the issue lies.

Could you please provide guidance on how to troubleshoot this problem and ensure proper connectivity between the frontend and backend APIs?

**Reply by S1LV3RJ1NX** [reply]
When you run locally, the current UI does not support collection creation and data source addition, everything is routed th' `local.metadata.yaml`. We are working to remove this limitation soon.

**Reply by HariHaranDFX** [reply]
Thank you for the clarification regarding the current limitations of the UI.

Could you please provide guidance on how to query against the collection? Specifically, I would like to understand the process and any necessary steps or configurations required to perform queries under the current setup.

**Reply by S1LV3RJ1NX** [reply]
once you start the docker-compose, the collection in `local.metadata.yaml` file is indexed and will be available to use on docsQA page. You can then query on the documents. Choose LLM (currently gemma:2b, but you can change the LLM under compose.env file), use different retrievers, change prompt if require and query.

**Reply by HariHaranDFX** [reply]
I understand but docsQA page not showing indexed collection, instead it's asking to create a new collection.

**Comment by S1LV3RJ1NX** [comment]
can you tell me the steps you have followed? @HariHaranDFX

**Reply by HariHaranDFX** [reply]
steps I have followed are after you updated for docker
- Created a venv 
- docker-compose --env-file compose.env up --build
- it installed and created five images frontend, backend, ollama, infinity, quadrant
- and ran the docker images
- and I used the URL provided in the browser
- infinity, backend for rag swagger pages opened ad frontend opened, url for quadrant not accessible but running, I tried to run the API in both infinity and backend all are working just fine and responded with the data injected as sample-credit cards

but when I wanted to use it query in frontend it's asking to created new collection as I mentioned before

**Comment by HariHaranDFX** [comment]
@S1LV3RJ1NX  please can you help me with this?

**Reply by S1LV3RJ1NX** [reply]
Not sure, what's the issue here, once you open the FE, it should start working, maybe model download is in progress. We are updating the conditions to account for such cases, FE will start only when all services are up and running and healthy. Keep an eye on #213

**Comment by HariHaranDFX** [comment]
Hi @S1LV3RJ1NX, I hope you are doing well.

I have setup Cognita using docker-compose its successfull, I am also running ollama and infinity in docker too all are up and running.
as mentioned i have updated models_config.yaml baseurl with the local url which is given below. i did not change anything else and tried adding data source its success, then created new collection and added the data source but got failed. please help me.

**models_config.yaml**

  - provider_name: local-ollama
    api_format: openai
    base_url: http://localhost:11434/api/tags/
    api_key_env_var: ''
    llm_model_ids:
      - 'qwen2:1.5b'
    embedding_model_ids: []

  - provider_name: local-infinity
    api_format: openai
    base_url: http://localhost:7997/
    api_key_env_var: ''
    llm_model_ids: []
    embedding_model_ids:
      - 'mixedbread-ai/mxbai-embed-large-v1'
      

**Error stack:**

cognita-backend   | INFO:    2024-06-25 13:22:01,691 - collection:create_collection:70 - Creating collection dfxwebsite...
cognita-backend   | INFO:    2024-06-25 13:22:01,720 - prismastore:acreate_collection:64 - Creating collection: {'name': 'dfxwebsite', 'description': None, 'embedder_config': {'model_config': {'name': 'local-infinity/mixedbread-ai/mxbai-embed-large-v1', 'type': None, 'parameters': None}, 'config': {}}}
cognita-backend   | INFO:    2024-06-25 13:22:01,747 - collection:create_collection:79 - Creating collection dfxwebsite on vector db...                                                                                                     
cognita-backend   | /virtualenvs/venv/lib/python3.11/site-packages/langchain_openai/embeddings/base.py:144: UserWarning: WARNING! check_embedding_ctx_length is not default parameter.                                                      
cognita-backend   |                     check_embedding_ctx_length was transferred to model_kwargs.
cognita-backend   |                     Please confirm that check_embedding_ctx_length is what you intended.          
cognita-backend   |   warnings.warn(
cognita-backend   | DEBUG:    2024-06-25 13:22:01,816 - qdrant:create_collection:44 - [Qdrant] Creating new collection dfxwebsite
cognita-backend   | DEBUG:    2024-06-25 13:22:01,816 - qdrant:create_collection:47 - [Qdrant] Embedding a dummy doc to get vector dimensions                                                                                               
cognita-backend   | ERROR:    2024-06-25 13:22:02,454 - collection:create_collection:106 - Failed to create collection
cognita-backend   | Traceback (most recent call last):
cognita-backend   |   File "/app/backend/server/routers/collection.py", line 80, in create_collection
cognita-backend   |     VECTOR_STORE_CLIENT.create_collection(                                                        
cognita-backend   |   File "/app/backend/modules/vector_db/qdrant.py", line 48, in create_collection
cognita-backend   |     partial_embeddings = embeddings.embed_documents(["Initial document"])                         
cognita-backend   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cognita-backend   |   File "/virtualenvs/venv/lib/python3.11/site-packages/langchain_openai/embeddings/base.py", line 508, in embed_documents
cognita-backend   |     return self._get_len_safe_embeddings(texts, engine=engine)
cognita-backend   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cognita-backend   |   File "/virtualenvs/venv/lib/python3.11/site-packages/langchain_openai/embeddings/base.py", line 324, in _get_len_safe_embeddings                                                                                      
cognita-backend   |     response = self.client.create(
cognita-backend   |                ^^^^^^^^^^^^^^^^^^^                                                                
cognita-backend   | TypeError: Embeddings.create() got an unexpected keyword argument 'check_embedding_ctx_length'

**Reply by S1LV3RJ1NX** [reply]
Can you take the latest pull and rebuild the image and check. This issue must have been solved.

**Comment by agutgutia1991** [comment]
Hi Hariharan,
Could you also share your email? Will invite you to the cognita evangelists
internal slack.
Thanks!

**Reply by HariHaranDFX** [reply]
Hi @agutgutia1991,

Thank you for the invitation! I’m excited to join the Cognita Evangelists internal Slack.

Here’s my email: "hariharanbabuwk@gmail.com"

Looking forward to connecting with everyone!

**Reply by S1LV3RJ1NX** [reply]
Sent the invite!

**Comment by mnvsk97** [comment]
Hello @HariHaranDFX 

Hope you are doing well. Are you still facing this issue?
